react/features/app/functions.web.js

```diff
@@ -0,0 +1,214 @@
+/* global APP, JitsiMeetJS, loggingConfig */
+import { isRoomValid } from '../base/conference';
+import { RouteRegistry } from '../base/navigator';
+import { Conference } from '../conference';
+import { WelcomePage } from '../welcome';
+
+import getTokenData from '../../../modules/tokendata/TokenData';
+import settings from '../../../modules/settings/Settings';
+
+import URLProcessor from '../../../modules/config/URLProcessor';
+import JitsiMeetLogStorage from '../../../modules/util/JitsiMeetLogStorage';
+
+// eslint-disable-next-line max-len
+import KeyboardShortcut from '../../../modules/keyboardshortcut/keyboardshortcut';
+
+const Logger = require('jitsi-meet-logger');
+const LogCollector = Logger.LogCollector;
+
+
+/**
+ * Gets room name and domain from URL object.
+ *
+ * @param {URL} url - URL object.
+ * @private
+ * @returns {{
+ *      domain: (string|undefined),
+ *      room: (string|undefined)
+ *  }}
+ */
+function _getRoomAndDomainFromUrlObject(url) {
+    let domain;
+    let room;
+
+    if (url) {
+        domain = url.hostname;
+        room = url.pathname.substr(1);
+
+        // Convert empty string to undefined to simplify checks.
+        if (room === '') {
+            room = undefined;
+        }
+        if (domain === '') {
+            domain = undefined;
+        }
+    }
+
+    return {
+        domain,
+        room
+    };
+}
+
+/**
+ * Gets conference room name and connection domain from URL.
+ *
+ * @param {(string|undefined)} url - URL.
+ * @returns {{
+ *      domain: (string|undefined),
+ *      room: (string|undefined)
+ *  }}
+ */
+export function _getRoomAndDomainFromUrlString(url) {
+    // Rewrite the specified URL in order to handle special cases such as
+    // hipchat.com and enso.me which do not follow the common pattern of most
+    // Jitsi Meet deployments.
+    if (typeof url === 'string') {
+        // hipchat.com
+        let regex = /^(https?):\/\/hipchat.com\/video\/call\//gi;
+        let match = regex.exec(url);
+
+        if (!match) {
+            // enso.me
+            regex = /^(https?):\/\/enso\.me\/(?:call|meeting)\//gi;
+            match = regex.exec(url);
+        }
+        if (match && match.length > 1) {
+            /* eslint-disable no-param-reassign, prefer-template */
+
+            url
+                = match[1] /* URL protocol */
+                + '://enso.hipchat.me/'
+                + url.substring(regex.lastIndex);
```

Comment:
1. The file react/features/app/functions.web.js is practically a copy of a 117 lines from react/features/app/functions.native.js with the additions of functions. As per the coding style, this is unacceptable.

2. The copied parts have been modified with incorrect indentations.